homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,144 @@
+"""Platform for light integration."""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all devices and setup the switch devices via config entry."""
+    devices = hass.data[DOMAIN]["homecontrol"].binary_switch_devices
+
+    entities = []
+    for device in devices:
+        for binary_switch in device.binary_switch_property:
+            entities.append(
+                DevoloSwitch(
+                    homecontrol=hass.data[DOMAIN]["homecontrol"],
+                    device_instance=device,
+                    element_uid=binary_switch,
+                )
+            )
+    async_add_entities(entities)
+
+
+class DevoloSwitch(SwitchDevice):
+    """Representation of an Awesome Light."""
+
+    def __init__(self, homecontrol, device_instance, element_uid):
+        """Initialize an devolo Switch."""
+        self._device_instance = device_instance
+
+        # Create the unique ID
+        self._unique_id = element_uid
+
+        self._homecontrol = homecontrol
+        self._name = self._device_instance.itemName
+        self._available = self._device_instance.is_online()
+
+        # Get the brand and model information
+        self._brand = self._device_instance.brand
+        self._model = self._device_instance.name
+
+        self._binary_switch_property = self._device_instance.binary_switch_property.get(
+            self._unique_id
+        )
+        self._is_on = self._binary_switch_property.state
+
+        if hasattr(self._device_instance, "consumption_property"):
+            self._consumption = self._device_instance.consumption_property.get(
+                self._unique_id.replace("BinarySwitch", "Meter")
+            ).current
+        else:
+            self._consumption = None
+
+        self.subscriber = None
+
+    async def async_added_to_hass(self):
+        """Call when entity is added to hass."""
+        self.subscriber = async_dispatcher_connect(
```

Comment:
We need to send a signal somewhere if we want to use the dispatch helper. Sending the signal is done with `dispatcher_send`.

Reply:
Use an own subscriber here as the updates comes from the used API and there is no need to send them within HA